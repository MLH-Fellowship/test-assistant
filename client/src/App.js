import React, { useEffect, useReducer } from 'react'
import { Router } from '@reach/router'

import '../node_modules/eos-icons/dist/css/eos-icons.css'
import './assets/scss/index.scss'

import Dashboard from "./components/Dashboard";
import Login from './pages/Login'

import Context from './modules/Context'
import ContextReducer from './modules/ContextReducer'

const initialState = {
  auth: false,
  errorCode: null
}

const App = () => {
  const [state, dispatch] = useReducer(ContextReducer, initialState)

  const userId = localStorage.getItem('id')

  useEffect(() => {
    window.addEventListener('storage', (e) => {
      if (localStorage.getItem('id')) {
        dispatch({
          type: 'AUTHENTICATE'
        })
      } else {
        dispatch({
          type: 'DEAUTHENTICATE'
        })
      }
    })
  }, [])

  useEffect(() => {
    if (userId) {
      dispatch({
        type: 'AUTHENTICATE'
      })
    } else {
      dispatch({
        type: 'DEAUTHENTICATE'
      })
    }
  }, [userId])

  return (
    <Context.Provider value={{ state, dispatch }}>
      <Router>
        <Dashboard path='/' />
        <Login path='/login' />
      </Router>
    </Context.Provider>
  );
}

export default App;
