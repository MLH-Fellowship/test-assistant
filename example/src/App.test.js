import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App component', () => {
  it('starts with a count of 0', () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 0');
  });

  it('increments count by 1 when the increment button is clicked', () => {
    const wrapper = shallow(<App />);
    const incrementBtn = wrapper.find('button.increment');
    incrementBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 1');
  });

  it('decrements count by 1 when the decrement button is clicked', () => {
    const wrapper = shallow(<App />);
    const decrementBtn = wrapper.find('button.decrement');
    decrementBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: -1');
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
