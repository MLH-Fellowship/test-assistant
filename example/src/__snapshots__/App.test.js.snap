// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
